go/vt/vtgate/vindexes/vschema.go

```diff
@@ -406,7 +460,7 @@ func buildKeyspaceReferences(vschema *VSchema, ksvschema *KeyspaceSchema) error
 		sourceT.addReferenceInKeyspace(keyspace.Name, t)
 
 		// Forbid reference chains.
-		for sourceT.Source != nil {
+		for sourceT.Source != nil || (sourceT.Type == TypeReference && sourceT.Source != nil) {
```

Comment:
Needed to preserve existing behavior, where a reference table may define another reference table as its `source`, as long as that `source` does not itself define a `source`.

Previously this was done by `buildKeyspaceGlobalTables`, but now it is moved here.

Reply:
hmm... doesn't the `for` check now basically say: `A || (A && B)`, which can be simplified to just `A`?

Reply:
You're right! Good catch